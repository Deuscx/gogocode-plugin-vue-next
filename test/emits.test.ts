import { resolve } from 'path'
import { describe, expect, it } from 'vitest'
import $ from 'gogocode'
import type { Node } from '@babel/types'

// TODO:
// 1. appendEmitsProp

describe('should', () => {
  it.only('remove setup return', () => {
    const ast = $.loadFile(resolve(__dirname, './examples/emits1.vue'), {
      parseOptions: { language: 'vue' },
    })
    const script = ast.find('<script></script>')
    const setup = script.find('{ setup() {$$$0} }')
    const emits = script.find('{ emits: $_$1 }')
    script.find('setup($_$,$_$1) {}').each((item) => {
      const match = item.match[1][0]
      const node = match.node
      const nodeType = node.type
      console.log(item.match[1][0])
      if (nodeType === 'Identifier') {
        // ep: setup(_, context)
        const contextName = match.value
        setup.each((item) => {
          item.replace(`${contextName}.emit`, 'emit')
        })
      }
      else if (nodeType === 'ObjectPattern') {
        // ep: setup({}, {emit})

      }
    })
    console.log(emits)
    // script.find('setup(){}').remove()
    expect(ast.generate()).toMatchSnapshot()
  })
})
