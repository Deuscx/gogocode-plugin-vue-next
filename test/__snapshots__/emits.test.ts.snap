// Vitest Snapshot v1

exports[`should > remove setup return 1`] = `
"
<template>
    <div>
      <p>{{ text }}</p>
      <button @click=\\"$emit('accepted')\\">
        OK
      </button>
    </div>
</template>

<script lang=\\"ts\\">
import { defineComponent } from 'vue'

export default defineComponent({
  props: ['text'],
  emits: ['accepted'],
  setup(_, { emit }) {
    function accept() {
      emit('accepted')
    }
    return {
      accept,
    };
  },
})
</script>
"
`;
